hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CombinedBlockCache.java

```diff
@@ -507,4 +507,18 @@ public int evictBlocksRangeByHfileName(String hfileName, long initOffset, long e
     return l1Cache.evictBlocksRangeByHfileName(hfileName, initOffset, endOffset)
       + l2Cache.evictBlocksRangeByHfileName(hfileName, initOffset, endOffset);
   }
+
+  @Override
+  public boolean waitForCacheInitialization(long timeout) {
+    return (this.l2Cache instanceof BucketCache)
```

Comment:
Rather than assuming l1 would always be ready and adding BucketCache knowledge here, we should delegate the decision to both l1 and l2, like:
`return this.l1Cache.waitForCacheInitialization(timeout) && this.l2Cache.waitForCacheInitialization(timeout);`